crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -2418,21 +2414,18 @@ impl ForkState {
             .collect();
 
         Resolution {
-            nodes: packages,
-            edges: dependencies,
+            nodes,
+            edges,
             pins: self.pins,
             markers: self.markers,
         }
     }
 }
 
-/// The resolution from one or more forks including the virtual packages and the edges between them.
-///
-/// Each package can have multiple versions and each edge between two packages can have multiple
-/// version specifiers to support diverging versions and requirements in different forks.
+/// The resolution from a single fork including the virtual packages and the edges between them.
 #[derive(Debug)]
 pub(crate) struct Resolution {
-    pub(crate) nodes: FxHashMap<ResolutionPackage, FxHashSet<Version>>,
+    pub(crate) nodes: FxHashMap<ResolutionPackage, Version>,
```

Comment:
Ah nice, because each resolution is only allowed one version of a package, and `Resolution` no longer needs to represent the union of all resolutions since we merge them right into the graph?

Reply:
yep